src/cozmo/world.py

```diff
@@ -345,13 +345,44 @@ def _recv_msg_object_tapped(self, evt, *, msg):
             return
         obj.dispatch_event(evt)
 
-    def _recv_msg_object_states(self, evt, *, msg):
+    def _recv_msg_connected_object_states(self, evt, *, msg):
+        # This is received on startup as a response to RequestConnectedObjects
         for object_state in msg.objects:
             obj = self._objects.get(object_state.objectID)
             if not obj:
                 obj = self._allocate_object_from_msg(object_state)
             if obj:
-                obj._handle_object_state(object_state)
+                obj._handle_connected_object_state(object_state)
+
+    def _recv_msg_located_object_states(self, evt, *, msg):
+        # This is received on startup as a response to RequestLocatedObjectStates
+        # It's also automatically sent from Engine whenever poses are rejiggered
+        updated_objects = set()
+        for object_state in msg.objects:
+            obj = self._objects.get(object_state.objectID)
+            if not obj:
+                obj = self._allocate_object_from_msg(object_state)
+            if obj:
+                obj._handle_located_object_state(object_state)
+            updated_objects.add(object_state.objectID)
+        # verify that all objects not received have invalidated poses
+        for id, obj in self._objects.items():
+            if (id not in updated_objects) and obj.pose.is_valid:
+                logger.warn("Object %s still has a valid pose but wasn't part of located_object_state" % obj)
```

Comment:
should use comma instead of percent to pass the format values `logger.warn('... %s ...', obj)`